scripts/pylib/twister/twisterlib.py

```diff
@@ -3310,16 +3310,16 @@ def add_tasks_to_queue(self, pipeline, build_only=False, test_only=False):
             if build_only:
                 instance.run = False
 
-            if test_only and instance.run:
-                pipeline.put({"op": "run", "test": instance})
-            else:
-                if instance.status not in ['passed', 'skipped', 'error']:
-                    logger.debug(f"adding {instance.name}")
-                    instance.status = None
+            if instance.status not in ['passed', 'skipped', 'error']:
```

Comment:
I am not sure if there is not a flaw in the logic for `status == 'error'`. Before, not excluded instances were repeated from "the beginning" i.e. the "cmake" stage. Therefore, if there was a build error there was no point in retrying the building. I am worried that with this change if you experience an error during execution, e.g. during flashing, it won't be repeated. I will test this out.

Reply:
I've run a test making the test fail due to a flashing error and losing connection with the board and it was properly retried. It is because flashing errors get "flash_error" status, which is different than "error". I scanned the twisterlib.py code and it seems status == "error" can only be obtained during cmake/building stages. I saw no places were error can be obtained during test execution.